elrond-wasm-debug/src/mandos_system/model/step/step_sc_call.rs

```diff
@@ -87,4 +95,78 @@ impl ScCallStep {
         self.expect = Some(expect);
         self
     }
+
+    /// Sets following fields based on the smart contract proxy:
+    /// - "to"
+    /// - "function"
+    /// - "arguments"
+    pub fn call<OriginalResult>(
+        mut self,
+        contract_call: ContractCall<DebugApi, OriginalResult>,
+    ) -> Self {
+        let (to_str, function, mandos_args) = process_contract_call(contract_call);
+        self = self.to(to_str.as_str());
+        self = self.function(function.as_str());
+        for arg in mandos_args {
+            self = self.argument(arg.as_str());
+        }
+        self
+    }
+
+    /// Sets following fields based on the smart contract proxy:
+    /// - "to"
+    /// - "function"
+    /// - "arguments"
+    /// - "expect"
+    ///     - "out"
+    ///     - "status" set to 0
+    pub fn call_expect<OriginalResult, ExpectedResult>(
+        mut self,
+        contract_call: ContractCall<DebugApi, OriginalResult>,
+        expect_value: ExpectedResult,
+    ) -> Self
+    where
+        OriginalResult: TopEncodeMulti,
+        ExpectedResult: CodecFrom<OriginalResult> + TopEncodeMulti,
+    {
+        self = self.call(contract_call);
+        self = self.expect(format_expect(expect_value));
+        self
+    }
+}
+
+/// Extracts
+/// - recipient,
+/// - endpoint name,
+/// - the arguments.
+pub(super) fn process_contract_call<OriginalResult>(
+    contract_call: ContractCall<DebugApi, OriginalResult>,
+) -> (String, String, Vec<String>) {
```

Comment:
I tend to avoid tuples when some of the fields have the same type, as it's easy to make a mistake and return/use them in the wrong order. I'd suggest a wrapper struct, something like "ContractCallRawString".

Reply:
you're right, but it's private and will stay so, it is just used internally for code deduplication